File: Structural/Decorator/Python/Structure/ConcreteDecoratorB.py
"""
ConcreteDecoratorB: Adds responsibilities to the component.
"""
from Decorator import Decorator


class ConcreteDecoratorB(Decorator):
    """Concrete decorator that adds behavior B."""
    
    def operation(self) -> str:
        """Add behavior B before calling the wrapped component."""
        return f"ConcreteDecoratorB({self.component.operation()})"

File: Creational/Decorator/Java/Structure/ConcreteDecoratorB.java
package Creational.Decorator.Java.Structure;

public class ConcreteDecoratorB extends Decorator {
    public ConcreteDecoratorB(Component wrappee) {
        super(wrappee);
    }

    @Override
    public String operation() {
        return wrappee.operation() + "!";
    }
}
